src/renderers/webgpu/utils/WebGPUUtils.js

```diff
@@ -113,13 +113,21 @@ class WebGPUUtils {
 		// TODO: Remove this check when Quest 34.5 is out
 		// https://github.com/mrdoob/three.js/pull/29221/files#r1731833949
 
-		if ( navigator.userAgent.includes( 'Quest' ) ) {
+		if ( typeof navigator !== 'undefined' ) {
```

Comment:
Could we remove this? It is inside a function that will never be executed by the adapter return null in this case.

Reply:
I guess the issue is that it's a compiler error, not a runtime error. I wish we knew what in the Next.js stack is doing compile-time checks on dependencies, and what its requirements are. Avoiding all references to browser globals is a pretty tall order for a large library relying heavily on Web APIs. :/

Reply:
The compiler throwing an error for a global variable inside a function seems strange to say the least, since a variable can be declared at the global level after the function declaration.

Reply:
Yeah, I'm not sure why... trying to reproduce the issue:

- https://github.com/mrdoob/three.js/issues/29916#issuecomment-2487126519

Reply:
I think it is this one:
https://react.dev/reference/react/StrictMode